unittests/test-contracts/svnn_ibc/svnn_ibc.hpp

```diff
@@ -50,46 +47,43 @@ CONTRACT svnn_ibc : public contract {
          return tp;
       }
 
-      /*
-
-      //discuss : compute merkle branch direction vs providing them as part of the proof
-
-      static std::vector<uint8_t> _get_directions(const uint64_t index, const uint64_t last_node_index){
+      //compute proof path
+      static std::vector<bool> _get_proof_path(const uint64_t c_leaf_index, uint64_t const c_leaf_count) {
 
-          std::vector<uint8_t> proof;
+         uint64_t leaf_index = c_leaf_index;
+         uint64_t leaf_count = c_leaf_count;
 
-          uint64_t c_index = index;
-          uint64_t layers_depth = calculate_max_depth(last_node_index) -1;
-          uint64_t c_last_node_index = last_node_index;
+         std::vector<bool> proof_path;
 
-          for (uint64_t i = 0; i < layers_depth; i++) {
-              if (c_last_node_index % 2) c_last_node_index+=1;
-              bool isLeft = c_index % 2 == 0 ? 0 : 1;
-              uint64_t pairIndex = isLeft ? c_index - 1 :
-                          c_index == last_node_index - 1 && i < layers_depth - 1 ? c_index :
-                          c_index + 1;
-              c_last_node_index/=2;
-              if (pairIndex < last_node_index) proof.push_back(isLeft);
-              c_index = c_index / 2;
-          }
-          return proof;
-      }
+         uint64_t layers_depth = calculate_max_depth(c_leaf_count) -1;
 
-      */
+         for (uint64_t i = 0; i < layers_depth; i++) {
+            bool isLeft = leaf_index % 2;
+            uint64_t pairIndex = isLeft ? leaf_index - 1 :
+                           leaf_index == leaf_count ? leaf_index :
+                           leaf_index + 1;

```

Comment:
Better use parentheses so the reader doesn't have to remember esoteric operator precedence.
```suggestion
            uint64_t pairIndex = isLeft ? leaf_index - 1 :
                           (leaf_index == leaf_count ? leaf_index : leaf_index + 1);

```